### DISCUSSION TITLE
Run only a small piece of code on the client side in a component that needs to run in client and server

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#19337] [Questions] [answered]
I'm sorry if this is a really simple question, I'm a bit new to ssr frameworks and nuxt, previously I was working only in shopify.

This code below runs on both the server and the client so I will get an error due to the window.  This is a simplified example but I cannot wrap the whole component with `<ClientOnly />`. What is the way to go about this, I looked at the api section but I wasn't finding anything about what the nuxt process is. Would the solution be to wrap it all in the mounted hook or is there a better way to go about doing this?

```js
<script setup>
const mql = window.matchMedia('(max-width: 1024px)')

function handleMqlChange(e) {/* ... */}
onMounted(() => mql.addEventListener('change', handleMqlChange)
</script>
```

**Comment by ndragun92** [comment, marked as answer]
Hello, one way of doing it is.

As well as there is `<client-only>` there is check for script like `if(process.client)`

```typescript
<script setup>
const mql = process.client ? window.matchMedia('(max-width: 1024px)') : null;

function handleMqlChange(e) {
  console.log('onChange');
}
onMounted(() => {
  console.log('Check if MQL is here', mql);
  mql?.addEventListener('change', handleMqlChange);
});
</script>
```

Small example attached

https://stackblitz.com/edit/github-k6smmh?file=app.vue

Hope it helps

**Reply by dbzx10299** [reply]
Aha `process.client` was what I needed, I didn't want the component to render only on the client. I looked all over the docs for something like this and I wasn't able to find anything

**Reply by ndragun92** [reply]
Hope it did solve what you did need. In case it did solve your issue please close this discussion. If not then someone else might approach

**Reply by dbzx10299** [reply]
For some reason this discussion does not have an accept answer otherwise usually I do mark it as soon as possible

**Reply by ndragun92** [reply]
@manniL could you maybe mark it as an accepted answer in this case?

**Reply by dbzx10299** [reply]
Theres no option here, usually there is, but for some reason there is no option here to mark it.

**Reply by TheAlexLichter** [reply]
Moved to "Questions" and done
